includes/class-newspack-popups-exporter.php

```diff
@@ -0,0 +1,263 @@
+	 * Store the errors of items that could not be exported.
+	 *
+	 * @var int[]
+	 */
+	private $errors;
+
+	/**
+	 * Constructor.
+	 */
+	public function __construct() {
+		$this->reset_results();
+	}
+
+	/**
+	 * Export prompts, segments and campaigns.
+	 *
+	 * @return array Exported data.
+	 */
+	public function export() {
+		$this->reset_results();
+		$exported_data = [
+			'prompts'   => $this->get_prompts(),
+			'segments'  => $this->get_segments(),
+			'campaigns' => $this->get_campaigns(),
+		];
+
+		return [
+			'totals' => $this->totals,
+			'errors' => $this->errors,
+			'data'   => $exported_data,
+		];
+	}
+
+	/**
+	 * Reset results for a new export
+	 *
+	 * @return void
+	 */
+	private function reset_results() {
+		$this->totals = [
+			'prompts'   => 0,
+			'segments'  => 0,
+			'campaigns' => 0,
+		];
+		$this->errors = [
+			'prompts'   => [],
+			'segments'  => [],
+			'campaigns' => [],
+			'terms'     => [],
+		];
+	}
+
+	/**
+	 * Get the prompts to be exported.
+	 *
+	 * Return only items that pass validation and add errors for the ones that don't.
+	 *
+	 * @return array
+	 */
+	private function get_prompts() {
+		$prompts        = [];
+		$stored_prompts = Newspack_Popups_Model::retrieve_popups( true );
+		foreach ( $stored_prompts as $stored_prompt ) {
+			$transformed = $this->prepare_prompt_for_export( $stored_prompt );
+			$val         = new Newspack\Campaigns\Schemas\Prompts( $transformed );
+			if ( $val->is_valid() ) {
+				$prompts[] = $transformed;
+				$this->totals['prompts']++;
+			} else {
+				$this->errors['prompts'][ $stored_prompt['id'] ] = $val->get_errors();
+			}
+		}
+
+		return $prompts;
+	}
+
+	/**
+	 * Get the Campaigns to be exported.
+	 *
+	 * Return only items that pass validation and add errors for the ones that don't.
+	 *
+	 * @return array
+	 */
+	private function get_campaigns() {
+		$campaigns        = [];
+		$stored_campaigns = $this->sanitize_campaign_groups( Newspack_Popups::get_groups() );
+		foreach ( $stored_campaigns as $stored_campaign ) {
+			$val = new Newspack\Campaigns\Schemas\Campaigns( $stored_campaign );
+			if ( $val->is_valid() ) {
+				$campaigns[] = $stored_campaign;
+				$this->totals['campaigns']++;
+			} else {
+				$this->errors['campaigns'][ $stored_campaign->term_id ] = $val->get_errors();
+			}
+		}
+		return $campaigns;
+	}
+
+	/**
+	 * Get the Segments to be exported.
+	 *
+	 * Return only items that pass validation and add errors for the ones that don't.
+	 *
+	 * @return array
+	 */
+	private function get_segments() {
+		$segments        = [];
+		$stored_segments = Newspack_Popups_Segmentation::get_segments();
+		foreach ( $stored_segments as $stored_segment ) {
+			$val = new Newspack\Campaigns\Schemas\Segments( $stored_segment );
+			if ( $val->is_valid() ) {
+				$segments[] = $this->prepare_segment_for_export( $stored_segment );
+				$this->totals['segments']++;
+			} else {
+				$this->errors['segments'][ $stored_segment->term_id ] = $val->get_errors();
```

Comment:
For this line, the following warning is logged: `PHP Warning:  Attempt to read property "term_id" on array`